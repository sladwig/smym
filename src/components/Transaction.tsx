import { ITransaction } from '../store/Transaction';
import React from 'react';
import './transaction.css';
import { Icon } from './Icon';
import { store } from '../store/Store';

interface IProps {
    transaction: ITransaction;
}

export const Transaction = ({ transaction }: IProps) => {
    const backgroundColor = store.getColorFor(transaction.description);
    return (
        <div className="transaction">
            <Icon name="calendar" noHover={true} />
            <span className="transaction-date">
                {transaction.date.getDay()} {months[transaction.date.getMonth()]}
            </span>
            {transaction.description && (
                <span className="transaction-description" style={{ backgroundColor }}>
                    {transaction.description}
                </span>
            )}
            <span style={{ flexGrow: 1 }} />
            <span className="balance">{transaction.value.toFixed(2)}</span>
        </div>
    );
};

const months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Okt', 'Nov', 'Dec'];
