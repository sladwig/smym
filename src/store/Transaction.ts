import { types } from 'mobx-state-tree';

export const Transaction = types.model('Transaction', {
    date: types.optional(types.Date, () => new Date()),
    description: '',
    value: 0,
});

export type ITransaction = typeof Transaction.Type;
