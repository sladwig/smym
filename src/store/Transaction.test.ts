import { Transaction } from './Transaction';

test('can be created', () => {
    const transaction = Transaction.create({ date: 1556447289344 });
    expect(transaction).toMatchSnapshot();
});
