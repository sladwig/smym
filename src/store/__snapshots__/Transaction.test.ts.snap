// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can be created 1`] = `
Object {
  "date": 1556447289344,
  "description": "",
  "value": 0,
}
`;
